### ISSUE TITLE
Can't training model with multi GPU

### ISSUE THREAD

**Issue by ChungNPH** [tensorflow/tensorflow#40137]
Hi, I try to train my model on Keras 2.3 and TF 2.2. I use tf.distribute.MirroredStrategy() and have issue #38500 and tf.keras.utils.multi_gpu_model but it was removed after 2020-04-01. I also face with OOM errors because i can't specify device to train. Please tell me how can i train model with multi-gpu and avoid OOM errors. Thank you so much!

**Comment by amahendrakar** [comment]
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the OOM issue. Thanks!

Also, please take a look at [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and check if limiting GPU memory growth resolves the issue. Thanks!

**Comment by ChungNPH** [comment]
My code:

batch_size = 16

data = ImageDataGenerator(rotation_range=20, zoom_range=0.15, width_shift_range=0.2, height_shift_range=0.2,
                          shear_range=0.15, horizontal_flip=True, fill_mode="nearest", validation_split=0.2)
traindata = data.flow_from_directory(directory="../image-docs-similarity/visual-image-search/data/",target_size=(380,380), batch_size=batch_size, subset='training')
testdata = data.flow_from_directory(directory="../image-docs-similarity/visual-image-search/data/", target_size=(380,380), batch_size=batch_size, subset='validation')

with tf.device('/cpu:0'):
    model = efn.EfficientNetB4(weights='imagenet')
    predictions = Dense(115, activation='softmax')(model.layers[-2].output)
    model = Model(inputs=model.input, outputs=predictions)

opt = SGD(lr=1e-2) #num_epoch=100
parallel_model = multi_gpu_model(model, gpus=2)
parallel_model.compile(optimizer=opt, loss='categorical_crossentropy', metrics=['acc',f1_m,precision_m, recall_m])
# checkpoint = ModelCheckpoint("vgg16_1.h5", monitor='val_acc', verbose=1, save_best_only=True, save_weights_only=False, mode='auto', period=1)
early = EarlyStopping(monitor='val_acc', min_delta=0, patience=20, verbose=1, mode='auto')

hist = parallel_model.fit_generator(steps_per_epoch=9501//batch_size,generator=traindata, validation_data= testdata, validation_steps=2324//batch_size,epochs=20,callbacks=[early])

I face OOM when increasing batch_size. It's work with <=8. I used both MirroredStrategy() and multi_gpu_model but they aren't work. I hope keras have another way to train with multi gpu. Does it have? thank you
